src/engine/mod.rs

```diff
@@ -0,0 +1,659 @@
+//! This module provides WarpGrapher servers, including supporting modules for
+//! configuration, GraphQL schema generation, resolvers, and interface to the
+//! database.
+
+use actix_web::web::Json;
+
+use super::error::{Error, ErrorKind};
+use config::{WarpgrapherConfig, WarpgrapherProp, WarpgrapherResolvers, WarpgrapherValidators};
+use context::{GraphQLContext, WarpgrapherRequestContext};
+use extensions::WarpgrapherExtensions;
+use juniper::http::GraphQLRequest;
+use log::{debug, error};
+use r2d2::Pool;
+use r2d2_cypher::CypherConnectionManager;
+use schema::{create_root_node, RootRef};
+use serde_json;
+use serde_json::json;
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::option::Option;
+use std::env::var_os;
+
+pub mod config;
+pub mod context;
+pub mod extensions;
+pub mod neo4j;
+pub mod objects;
+mod resolvers;
+pub mod schema;
+mod visitors;
+
+#[allow(clippy::borrowed_box)]
+fn graphql_error(err: &Box<dyn std::error::Error + Send + Sync>) -> Error {
+    match serde_json::to_string(&json!({ "message": format!("{}", err) })) {
+        Ok(s) => Error::new(ErrorKind::GraphQLError(s), None),
+        Err(e) => {
+            error!("Failed to serialize error object:  {:#?}", e);
+            Error::new(ErrorKind::GraphQLError("INTERNAL SERVER ERROR".to_string()), None)
+        }
+    }
+}
+
+impl WarpgrapherRequestContext for () {
+    fn new() {}
+}
+
+/// A WarpGrapher GraphQL server.
```

Comment:
Need to update the docs (Server -> Engine).

Reply:
Docs updated to reflect Server->Engine change. Also undid change we made to disable doc code while dev.